src/common/file_location.c

```diff
@@ -137,58 +157,122 @@ char *dt_loc_find_install_dir(const char *suffix, const char *searchname)
 
 int dt_loc_init_tmp_dir(const char *tmpdir)
 {
-  darktable.tmpdir = dt_loc_init_generic(tmpdir, g_get_tmp_dir());
+  darktable.tmpdir = dt_loc_init_generic(tmpdir, NULL, g_get_tmp_dir());
+  dt_check_opendir("darktable.tmpdir", darktable.tmpdir, FALSE);
   if(darktable.tmpdir == NULL) return 1;
   return 0;
 }
 
 void dt_loc_init_user_cache_dir(const char *cachedir)
 {
   char *default_cache_dir = g_build_filename(g_get_user_cache_dir(), "darktable", NULL);
-  darktable.cachedir = dt_loc_init_generic(cachedir, default_cache_dir);
+  darktable.cachedir = dt_loc_init_generic(cachedir, NULL, default_cache_dir);
+  dt_check_opendir("darktable.cachedir", darktable.cachedir, TRUE);
   g_free(default_cache_dir);
 }
 
-void dt_loc_init_plugindir(const char *plugindir)
+void dt_loc_init_plugindir(const char* application_directory, const char *plugindir)
 {
 #if defined(__APPLE__) || defined(_WIN32)
   char *suffix = g_build_filename("lib", "darktable", NULL);
   char *directory = dt_loc_find_install_dir(suffix, darktable.progname);
   g_free(suffix);
-  darktable.plugindir = dt_loc_init_generic(plugindir, directory ? directory : DARKTABLE_LIBDIR);
+  darktable.plugindir = dt_loc_init_generic(plugindir, application_directory, directory ? directory : DARKTABLE_LIBDIR);
+  dt_check_opendir("darktable.plugindir", darktable.plugindir, TRUE);
   g_free(directory);
 #else
-  darktable.plugindir = dt_loc_init_generic(plugindir, DARKTABLE_LIBDIR);
+  darktable.plugindir =  dt_loc_init_generic(plugindir, application_directory, DARKTABLE_LIBDIR);
+  dt_check_opendir("darktable.plugindir", darktable.plugindir, TRUE);
 #endif
 }
 
-void dt_loc_init_localedir(const char *localedir)
+void dt_check_opendir(const char* text, const char* directory, gboolean exit_on_error)
+{
+  if (!directory) {
+    printf("directory for %s has not been set.\n", text);
+    exit(EXIT_FAILURE);
+  } 
+
+  DIR* dir = opendir(directory);
+  if (dir) {
+    dt_print(DT_DEBUG_DEV, "%s: %s\n", text, directory);
+    closedir(dir);
+  } 
+  else 
+  {
+    if ( errno == EACCES ) 
+    {
+      printf("Permission denied.\n");
```

Comment:
All errors should be sent to stderr -> fprintf(stderr, ...)

And message should not use capital-letter (sorry that's the current convention) and last but not least the messages should be made translatable _(" ... ");.

Reply:
Do we really want to translate error messages printed on the console?

Reply:
@parafin : I was sure that we already did that, but in fact no, no error messages on the console are translated and this is even better indeed. @codingdave : forget about the translation part, @parafin is correct we don't want that. Thanks.